### PR TITLE
[docs] Update upgrading Expo SDK walkthrough

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Fix ENG-13921

# How

- Update upgrading Expo SDK walkthrough doc to add SDK 52 info, link to changelog post, and deprecate SDK 50.

# Test Plan

CleanShot 2024-11-12 at 17 59 53

CleanShot 2024-11-12 at 18 00 00

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx [hunk 1]
```diff
@@ -30,7 +30,7 @@ Install the new version of the Expo package:
     '$ npm install expo@latest',
     '',
     '# Install a specific SDK version (for example, SDK 51)',
```
Comment (reichhartd) [inline docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx:32]:
```suggestion
    '# Install a specific SDK version (for example, SDK 52)',
```

File: docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx [hunk 2]
```diff
@@ -44,7 +44,7 @@ Install the new version of the Expo package:
     '$ yarn add expo@latest',
     '',
     '# Install a specific SDK version (for example, SDK 51)',
```
Comment (reichhartd) [inline docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx:46]:
```suggestion
    '# Install a specific SDK version (for example, SDK 52)',
```

File: docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx [hunk 3]
```diff
@@ -79,16 +79,18 @@ If you manage your own native projects, you will need to manually apply any chan
...
+### SDK 52
+
+[Blog Post](https://expo.dev/changelog/2024/11-12-sdk-52)
+
 ### SDK 51
 
 [Blog post](https://expo.dev/changelog/2024/05-07-sdk-51)
 
-### SDK 50
+#### SDK 50
```
Comment (reichhartd) [inline docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx:90]: Was the change intentional?

```suggestion
### Deprecated SDK Version Changelogs

#### SDK 50
```

File: docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx (hunk 1 above)
Comment (amandeepmittal) [reply docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx:32]: Good catch!

File: docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx (hunk 3 above)
Comment (amandeepmittal) [reply docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx:90]: Yes, the last SDK version is deprecated in every SDK release cycle.

Comment (reichhartd) [reply docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx:90]: No, I was referring to removing the heading “Deprecated SDK Version Changelogs” 😄

Comment (amandeepmittal) [reply docs/pages/workflow/upgrading-expo-sdk-walkthrough.mdx:90]: Ah yes, my bad, sorry. You are right, the heading wasn't supposed to be removed. I'll add it back.

**Review by Simek — APPROVED** [review]

**Pull request merged**